presto-delta/src/main/java/com/facebook/presto/delta/DeltaClient.java

```
@@ -70,72 +77,101 @@ public DeltaClient(HdfsEnvironment hdfsEnvironment)
-                    format("Delta table %s has unsupported data format: %s. Currently only Parquet data format is supported", schemaTableName, format));
+        if (snapshot instanceof SnapshotImpl) {
+            String format = ((SnapshotImpl) snapshot).getMetadata().getFormat().getProvider();
+            if (!PARQUET.name().equalsIgnoreCase(format)) {
+                throw new PrestoException(DELTA_UNSUPPORTED_DATA_FORMAT,
+                        format("Delta table %s has unsupported data format: %s. Currently only Parquet data format is supported", schemaTableName, format));
+            }
         }
-
-        return Optional.of(new DeltaTable(
-                schemaTableName.getSchemaName(),
-                schemaTableName.getTableName(),
-                tableLocation,
-                Optional.of(snapshot.getVersion()), // lock the snapshot version
-                getSchema(schemaTableName, metadata)));
+        return snapshot;
     }
 
     /**
      * Get the list of files corresponding to the given Delta table.
      *
      * @return Closeable iterator of files. It is responsibility of the caller to close the iterator.
      */
-    public CloseableIterator<AddFile> listFiles(ConnectorSession session, DeltaTable deltaTable)
+    public CloseableIterator<FilteredColumnarBatch> listFiles(ConnectorSession session, DeltaTable deltaTable)
     {
+        requireNonNull(deltaTable, "deltaTable is null");
         checkArgument(deltaTable.getSnapshotId().isPresent(), "Snapshot id is missing from the Delta table");
-        Optional<DeltaLog> deltaLog = loadDeltaTableLog(
-                session,
+        Optional<Engine> deltaEngine = loadDeltaTableClient(session,
                 new Path(deltaTable.getTableLocation()),
                 new SchemaTableName(deltaTable.getSchemaName(), deltaTable.getTableName()));
+        if (!deltaEngine.isPresent()) {
+            throw new PrestoException(GENERIC_INTERNAL_ERROR,
```

Please throw a system error here.  GENERIC_INTERNAL_ERROR is an error for anything that is not categorized, so we shouldn't be intentionally throwing it.

---

Done